pkg/executor/kubernetes.go

```diff
@@ -358,7 +359,21 @@ func (k8s *k8s) Execute(command string) (TaskHandle, error) {
 	}
 
 	// Best effort potential way to check if binary is started properly.
-	taskHandle.Wait(100 * time.Millisecond)
+	select {
+	case <-taskWatcher.failed:
+		// Pod was started but shell failed to launch requested process.
+		log.Errorf("K8s executor: task %q on pod %s failed", taskHandle.command, taskHandle.podName)
+		break
+	case <-taskWatcher.deleted:
+		// Pod was deleted before it even failed. It may happen when image is not available or communication with Docker fails.
+		log.Errorf("K8s executor: pod %s was deleted immediately after being started", taskHandle.podName)
```

Comment:
Could we work something our with this error? Maybe `pod %q could not be started by kubernetes`? To distinguish the fact that pod can fail-immedietatly-after-run and fail-not-even-started.
The 'pod deleted immedietaly' notion is not something in Kubernetes terminology, does it?